src/main/java/nl/tudelft/scrumbledore/CollisionsLevelModifier.java

```diff
@@ -268,21 +281,22 @@ public void detectPlayerBubble(Level level, double delta) {
   public void detectBubbleEnemy(Level level, double delta) {
     ArrayList<NPC> enemies = level.getNPCs();
     ArrayList<Fruit> fruits = level.getFruits();
+    ArrayList<Bubble> bubbles = level.getBubbles();
 
-    if (level.getBubbles().size() > 0 && enemies.size() > 0) {
+    if (bubbles.size() > 0 && enemies.size() > 0) {
       for (int i = 0; i < enemies.size(); i++) {
-        for (int j = 0; j < level.getBubbles().size(); j++) {
 
+        for (int j = 0; j < bubbles.size(); j++) {
           // Temp fix to prevent race condition
           if (enemies.size() != i
-              && enemies.get(i).inBoxRangeOf(level.getBubbles().get(j), Constants.COLLISION_RADIUS)
-              && new Collision(level.getBubbles().get(j), enemies.get(i), delta).colliding()) {
-            level.getBubbles().remove(j);
-            Fruit newFruit = new Fruit(enemies.get(i).getPosition().clone(), new Vector(
-                Constants.BLOCKSIZE, Constants.BLOCKSIZE));
-            // Adding a new Fruit element in place of where the enemy died.
+              && enemies.get(i).inBoxRangeOf(bubbles.get(j), Constants.COLLISION_RADIUS)
+              && new Collision(bubbles.get(j), enemies.get(i), delta).colliding()) {            
+            
+            // The enemy gets removed and a new encapsulated enemy will appear.
             enemies.remove(i);
-            fruits.add(newFruit);
+            bubbles.get(j).setHasNPC(true);
```

Comment:
I think you should check whether this Bubble does not already have an NPC, currently this implementation allows for a Bubble to encapsulated multiple enemies.